packages/components/navlist/src/NavList/NavList.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { cx } from 'emotion';
+import {
+  type CommonProps,
+  type PolymorphicProps,
+  type PolymorphicComponent,
+  type ExpandProps,
+} from '@contentful/f36-core';
+import { getStyles } from './NavList.styles';
+
+const NAV_LIST_DEFAULT_TAG = 'div';
+
+export interface NavListInternalProps extends CommonProps {
+  children: React.ReactNode;
+  /**
+   * @default div
+   */
+  as?: 'nav' | 'div';
+}
+
+export type NavListProps<
+  E extends React.ElementType = typeof NAV_LIST_DEFAULT_TAG,
+> = PolymorphicProps<NavListInternalProps, E>;
+
+function _NavList<E extends React.ElementType = typeof NAV_LIST_DEFAULT_TAG>(
+  props: NavListProps<E>,
+  ref: React.Ref<any>,
+) {
+  const styles = getStyles();
```

Comment:
I know we use this pattern a lot, but I don't see a reason to recompute the styles here on every render, wdyt?

Reply:
This is so the css class is only generated when the component itself is rendered, otherwise just by importing the file it would create the class, bloating the bundle.
And IIRC emotion handles caching this 🤔